lib/build.gradle

```diff
@@ -104,10 +105,14 @@ dependencies {
 	compatKtLint0Dot49Dot0CompileAndTestOnly 'com.pinterest.ktlint:ktlint-rule-engine:0.49.0'
 	compatKtLint0Dot49Dot0CompileAndTestOnly 'com.pinterest.ktlint:ktlint-ruleset-standard:0.49.0'
 	compatKtLint0Dot49Dot0CompileAndTestOnly 'org.slf4j:slf4j-api:2.0.0'
-	// ktlint latest supported version
+	// ktlint previous supported version
 	compatKtLint0Dot50Dot0CompileAndTestOnly 'com.pinterest.ktlint:ktlint-rule-engine:0.50.0'
 	compatKtLint0Dot50Dot0CompileAndTestOnly 'com.pinterest.ktlint:ktlint-ruleset-standard:0.50.0'
 	compatKtLint0Dot50Dot0CompileAndTestOnly 'org.slf4j:slf4j-api:2.0.0'
+	// ktlint latest supported version
+	compatKtLint1Dot0Dot0CompileAndTestOnly 'com.pinterest.ktlint:ktlint-rule-engine:1.0.0'
```

Comment:
Not a review of your PR, I just wanted to say that I'm happy to see that the version compatibility Gradle plugin is being put to good use :)

Reply:
😄 I'd made it workable in #1517.

Reply:
Yeah, the plugin tries to impose as little configuration as possible, to avoid hard-to-resolve clashes with other plugins.

Another way to solve the test situation is to use the JDK versions in the compatibility matrix, as outlined here: https://github.com/davidburstrom/version-compatibility-gradle-plugin/blob/6ec1b14d954b83739108339148fb15bfda786733/example/build.gradle.kts#L31

Build caching improves if Gradle is always executing with a fixed JDK version, and pulls in toolchains as needed to compile or test against given JDK versions.